pasqal_cloud/batch.py

```diff
@@ -146,7 +164,16 @@ def declare_complete(
                     self.id,
                 )
             for job_rsp in batch_rsp["jobs"]:
-                self.jobs[job_rsp["id"]] = Job(**job_rsp)
+                job_rsp_obj = Job(**job_rsp)
+                # iterate through the ordered_job list attribute of Batch and find the
+                # index of the job received in the response to replace with updated data
+                job_index = [
+                    job_rsp_obj.id == job.id for job in self.ordered_jobs
```

Comment:
The order should be consistent no ?

Reply:
Should be yes, if it send back all the jobs of the Batch I guess can just override the ordered jobs values by the response ?

Reply:
Can we do this without having to build the whole list first? Something like 

```python
job_index = next(n for n, job in enumerate(self.ordered_jobs) if job.id == job_rsp_obj.id)
```

Reply:
The override is a bad idea ?

Reply:
I don't see why it wouldn't and we could get rid of building the full list as asked by James